cpu/numactl.py

```diff
@@ -75,26 +76,47 @@ def setUp(self):
         build.make(self.sourcedir)
 
         interfaces = netifaces.interfaces()
-        interface = self.params.get("interface")
-        if interface not in interfaces:
-            self.cancel("%s interface is not available" % interface)
-        self.iface = interface
+        self.iface = self.params.get("interface", default="")
+        if self.iface not in interfaces:
+            self.cancel("%s interface is not available" % self.iface)
         local = LocalHost()
         self.networkinterface = NetworkInterface(self.iface, local)
         self.ipaddr = self.params.get("host_ip", default="")
         self.netmask = self.params.get("netmask", default="")
-        self.peer = self.params.get("peer_ip")
+        self.peer = self.params.get("peer_ip", default="")
         if not self.peer:
             self.cancel("No peer provided")
+        if self.networkinterface.is_link_up() is not True:
+            self.cancel("Interface is not up")
```

Comment:
if the interfaces are not up, why cann't we try to bring it up. I feel we have the methods to bring the interfaces up, so just calling them will do test will run. just my thoughts